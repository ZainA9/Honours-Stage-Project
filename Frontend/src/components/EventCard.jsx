import React from 'react';

const categoryImages = {
  Technology: 'https://source.unsplash.com/featured/?technology,conference',
  Music: 'https://source.unsplash.com/featured/?music,concert',
  Food: 'https://source.unsplash.com/featured/?food,festival',
  Arts: 'https://source.unsplash.com/featured/?art,painting',
  Sports: 'https://source.unsplash.com/featured/?sports,fitness',
  Education: 'https://source.unsplash.com/featured/?education,classroom',
};

export default function EventCard({ event }) {
  const imageUrl = categoryImages[event.categories?.[0]] || 'https://source.unsplash.com/featured/?event';

  return (
    <div className="bg-white rounded-xl shadow overflow-hidden">
      <img src={imageUrl} alt={event.name} className="h-52 w-full object-cover" />
      <div className="p-4">
        <span className="inline-block text-sm bg-indigo-100 text-indigo-600 px-2 py-0.5 rounded-full mb-2">
          {event.categories?.[0]}
        </span>
        <h4 className="font-semibold text-lg">{event.name}</h4>
        <p className="text-sm text-gray-500 my-2">📅 {new Date(event.date).toLocaleDateString()}</p>
        <p className="text-sm text-gray-500 mb-4">📍 {event.location}</p>
        <button className="text-indigo-600 font-semibold">Get Tickets</button>
      </div>
    </div>
  );
}
