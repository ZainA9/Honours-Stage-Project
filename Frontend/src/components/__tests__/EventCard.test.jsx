import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import EventCard from '../EventCard';

describe('EventCard Snapshot', () => {
  it('matches the snapshot', () => {
    const mockEvent = {
      id: '123',
      name: 'Tech Meetup',
      location: 'City Center',
      date: '2024-05-10',
      categories: ['tech'],
      imageUrl: '',
    };

    const tree = renderer
      .create(
        <MemoryRouter>
          <EventCard event={mockEvent} />
        </MemoryRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
