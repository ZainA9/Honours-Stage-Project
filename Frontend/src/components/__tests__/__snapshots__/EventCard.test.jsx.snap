// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventCard Snapshot matches the snapshot 1`] = `null`;
